zipline/data/minute_bars.py

```diff
@@ -48,6 +50,99 @@ class BcolzMinuteWriterColumnMismatch(Exception):
     pass
 
 
+class MinuteBarReader(with_metaclass(ABCMeta)):
+
+    @abstractproperty
+    def last_available_dt(self):
+        """
+        Returns
+        -------
+        dt : pd.Timestamp
+            The last minute for which the reader can provide data.
+        """
+        pass
+
+    @abstractproperty
+    def first_trading_day(self):
+        """
+        Returns
+        -------
+        dt : pd.Timestamp
+            The first trading day (session) for which the reader can provide
+            data.
+        """
+        pass
+
+    @abstractmethod
+    def get_value(self, sid, dt, field):
+        """
+        Retrieve the value at the given coordinates.
+
+        Parameters
+        ----------
+        sid : int
+            The asset identifier.
+        dt : datetime64-like
+            The minute label for the desired data point.
+        field : string
+            The OHLVC name for the desired data point.
+
+        Returns
+        -------
+        value : float|int
+            The value at the given coordinates, `float` for OHLC, `int` for
+            'volume'.
+        """
+
+    @abstractmethod
+    def get_last_traded_dt(self, asset, dt):
```

Comment:
I believe the original distinction was that `get_value` was returning the pricing values, i.e. the trade bar floats/ints, and this function was a helper to find the right dt to seek into when doing forward filling; with the distinction that this returns the location of data as opposed to the value.

I am not opposed to merging `last_traded` into `get_value`, but there is a slight semantic difference. Also, some care may need to be taken when changing the code since this method is used internally by the data portal when returning values for `price`.